### PR TITLE
Add Python 3.7

### LINKED ISSUE #72: Add Python 3.7
I'd love to be able to build my docs for a Python 3.7-only project on RTD.

Since you're using `pyenv` from git, is it possible to add Python 3.7 to the rtd/build 3.x line of images, or will we have to wait until you switch to the 18.04-based image?

### PR DESCRIPTION
**Pull request by browniebroke**
Fixes #72 and based on #74 

- Add Python 3.7 to the Docker image
- Update README

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,10 +11,12 @@ ENV LANG C.UTF-8
 ENV PYTHON_VERSION_27 2.7.14
 ENV PYTHON_VERSION_35 3.5.5
 ENV PYTHON_VERSION_36 3.6.4
-ENV CONDA_VERSION 4.4.10
+ENV PYTHON_VERSION_37 3.7.0
```
Comment (browniebroke) [inline Dockerfile]: Note from [Python 3.7 release schedule](https://www.python.org/dev/peps/pep-0537/):

> 3.7.1 final: 2018-10-18 (expected)

We might want to hold this one for a bit?

Comment (humitos) [reply Dockerfile]: I suppose, yes.

Comment (browniebroke) [reply Dockerfile]: Ok! I'll update this PR when 3.7.1 gets released

**Comment by webknjaz** [comment]
@humitos this PR now points to Python 3.7.1
I think it's ready to be merged :)

**Comment by humitos** [comment]
Thanks!

**Comment by webknjaz** [comment]
@humitos is it possible to add this to latest?

**Comment by humitos** [comment]
@webknjaz we are trying to make our releases of docker images available in shorter times, but we could not get there yet.

I suppose this won't be available in the couple next weeks at least.

Is this blocking your docs to be built?

**Comment by webknjaz** [comment]
Not really, but nice to have :)

**Comment by humitos** [comment]
@stblassitude the import is a warning. I think your problem is because the f-string. You could use Py3.6 with `latest` docker image for that and it should work.

**Comment by humitos** [comment]
> Are there any updates to the availability for a python 3.7 friendly image?

We don't have a ETA yet :(

If you strictly need py3.7 you could use `conda` as a workaround for now. I don't really like this, but it's a possibility at the moment.

**Comment by vEpiphyte** [comment]
@humitos our requirement for 3.7 is strict.  I could test out the anaconda workflow to see if that would work as an alternative for now.

**Comment by vEpiphyte** [comment]
tl;dr:

1. Follow https://docs.readthedocs.io/en/latest/conda.html for setting up conda
2. Use https://conda.io/docs/user-guide/tasks/manage-environments.html to define your YAML file you are using for RTD
3. Check your RTD builds to ensure they are working properly with the correct python version and are installing the packages you specify.

----

Using the following patch to my project I was able to use 3.7 to perform a docs build (my build requires jupyter installed for docs derived from notebooks).  This resulted in docs successfully building in the RTD environment while using Python 3.7 in order to execute the build environment.

```
commit baf5a16145cea33a4021d2d7c1ac8fbc5e00f01d
Author: epiphyte <epiphyte@vertex.link>
Date:   Tue Dec 18 02:09:51 2018 +0000

    Try conda

diff --git a/environment_docs.yml b/environment_docs.yml
new file mode 100644
index 00000000..bd90cb73
--- /dev/null
+++ b/environment_docs.yml
@@ -0,0 +1,9 @@
+# Environment file for conda for readthedocs
+# This is to be used until rtd supports python 3.7 natively
+name: synapse-docs
+dependencies:
+  - python=3.7
+  - pip:
+    - sphinx
+    - jupyter
+    - sphinx-rtd-theme
\ No newline at end of file
diff --git a/readthedocs.yml b/readthedocs.yml
index 56264fe5..b5358b9b 100644
--- a/readthedocs.yml
+++ b/readthedocs.yml
@@ -1,10 +1,4 @@
 # .readthedocs.yml
 
-build:
-  image: 4.0rc1
-
-python:
-  version: 3.7
-  pip_install: true
-  extra_requirements:
-    - docs
+conda:
+    file: environment_docs.yml
```
